### PR TITLE
Add coverage profile retrieval to log dumps

### PR DESCRIPTION
**Pull request by Katharine**
**What this PR does / why we need it**:

This PR enables `kubetest --dump` to retrieve coverage profiles generated by #67971 when using the default `bash` provider (and specifically targeting GCE for the time being).

For kubelet, this is accomplished by simply dumping the coverage file into /var/log and adding it to the retrieval list.

For other components, we use `docker exec` to peek into the containers running the components. It is easier to do this than to mount a volume containing the coverage file due to the way coverage files are written, and executing directly using kubernetes (e.g. via `kubectl exec`) won't work because some components (e.g. apiserver) disable this capability. We know using docker to be viable on our initial targets for this feature.

Finally, it removes the assumption that everything we wish to retrieve will end in .log, because the coverage files don't.

/sig testing
/assign @spiffxp 

```release-note
NONE
```

### CODE REVIEW

**Comment by Katharine** [comment]
/retest

🙄

**Comment by spiffxp** [comment]
/lgtm
/assign @roberthbailey @mikedanese 
for /approve

File: cluster/gce/gci/configure-helper.sh [hunk 1]
```diff
@@ -1160,6 +1160,7 @@ function start-kubelet {
   local -r kubelet_env_file="/etc/default/kubelet"
   local kubelet_opts="${KUBELET_ARGS} ${KUBELET_CONFIG_FILE_ARG:-}"
   echo "KUBELET_OPTS=\"${kubelet_opts}\"" > "${kubelet_env_file}"
+  echo "KUBE_COVERAGE_FILE=\"/var/log/kubelet.cov\"" >> "${kubelet_env_file}"
```
Comment (roberthbailey) [inline cluster/gce/gci/configure-helper.sh]: It looks like this is going to enable coverage unconditionally. Is this something that should be enabled in production clusters or just in testing clusters?

@dchen1107 @abgworrall

Comment (Katharine) [reply cluster/gce/gci/configure-helper.sh]: It won't; coverage needs to be enabled at compile-time. This only affects where coverage will be dumped to if coverage is already enabled.

**Comment by Katharine** [comment]
I'm guessing something's borked. These are all the same failure.

image

**Comment by spiffxp** [comment]
/lgtm cancel
let's stop the /retest spam until we figure out what is up with gke

**Comment by Katharine** [comment]
48 retests later, it passed!

**Comment by spiffxp** [comment]
/lgtm
going to see whether I can get this into v1.12

**Comment by spiffxp** [comment]
/milestone v1.12
/priority critical-urgent
I'll leave the `/hold cancel` to @tpepper

**Comment by tpepper** [comment]
We got reasonable improvement in CI signal today, so I'm letting a couple more PR's in.

/hold cancel

**Pull request merged**